apps/openmw/mwmechanics/character.cpp

```diff
@@ -419,6 +419,47 @@ void CharacterController::refreshJumpAnims(const WeaponInfo* weap, JumpingState
     }
 }
 
+bool CharacterController::onOpen()
+{
+    if (mPtr.getTypeName() == typeid(ESM::Container).name())
+    {
+        if (!mAnimation->hasAnimation("containeropen"))
+            return true;
+
+        bool playing = mAnimation->isPlaying("containeropen");
+        if (playing)
+            return false;
+
+        playing = mAnimation->isPlaying("containerclose");
```

Comment:
Can we avoid this kind of variable re-use please? Doesn't look like we need a variable in the first place. The function call can go directly into the condition of the if statement.